### PR TITLE
Added Order storage

### PR DESCRIPTION
**Pull request by yeodonghan**
Added getValue() for Price, Constructor for Order (inclusive of status and schedule in parameter)

### CODE REVIEW

**Comment by yeodonghan** [comment]
@EugeneTeu I have changed the private constructor to public for Phone class hence the conflict.

File: src/main/java/seedu/address/model/OrderBook.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
  * Wraps all data at the order-book level
  * Duplicates are not allowed (by .isSameOrder comparison)
  */
-public class OrderBook implements ReadOnlyDataBook<Order> {
+public class OrderBook implements ReadOnlyOrderBook {
```
Comment (zhixianggg) [inline src/main/java/seedu/address/model/OrderBook.java]: could you change it back to ReadOnlyDataBook<Order>? o-o

File: src/main/java/seedu/address/storage/JsonOrderBookStorage.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    public JsonOrderBookStorage(Path filePath) {
+        this.filePath = filePath;
+    }
+
+    public Path getOrderBookFilePath() {
+        return filePath;
+    }
+
+    @Override
+    public Optional<ReadOnlyOrderBook> readOrderBook() throws DataConversionException {
```
Comment (zhixianggg) [inline src/main/java/seedu/address/storage/JsonOrderBookStorage.java]: could you change this to ReadOnlyDataBook<Order>? rationale is somewhere along the way, we would have to generalize and refactor as much as possible in our Model class

File: src/main/java/seedu/address/model/OrderBook.java [hunk 3]
```diff
@@ -12,7 +12,7 @@
  * Wraps all data at the order-book level
  * Duplicates are not allowed (by .isSameOrder comparison)
  */
-public class OrderBook implements ReadOnlyDataBook<Order> {
+public class OrderBook implements ReadOnlyDataBook {
 
     private final UniqueOrderList orders;
```
Comment (zhixianggg) [inline src/main/java/seedu/address/model/OrderBook.java]: can it implement ReadOnlyDataBook<Order>?

File: src/main/java/seedu/address/storage/JsonOrderBookStorage.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    public Path getOrderBookFilePath() {
+        return filePath;
+    }
+
+    @Override
+    public Optional<ReadOnlyDataBook> readOrderBook() throws DataConversionException {
+        return readOrderBook(filePath);
+    }
```
Comment (zhixianggg) [inline src/main/java/seedu/address/storage/JsonOrderBookStorage.java]: ReadOnlyDataBook<Order>

Comment (zhixianggg) [inline src/main/java/seedu/address/storage/JsonOrderBookStorage.java]: can it be changed to ReadOnlyDataBook <Order> ?

File: src/main/java/seedu/address/storage/JsonOrderBookStorage.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    /**
+     * Similar to {@link #readOrderBook()}.
+     *
+     * @param filePath location of the data. Cannot be null.
+     * @throws DataConversionException if the file is not in the correct format.
+     */
+    public Optional<ReadOnlyDataBook> readOrderBook(Path filePath) throws DataConversionException {
+        requireNonNull(filePath);
```
Comment (zhixianggg) [inline src/main/java/seedu/address/storage/JsonOrderBookStorage.java]: can it be changed to ReadOnlyDataBook <Order> ?

File: src/main/java/seedu/address/storage/JsonOrderBookStorage.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            logger.info("Illegal values found in " + filePath + ": " + ive.getMessage());
+            throw new DataConversionException(ive);
+        }
+    }
+
+    @Override
+    public void saveOrderBook(ReadOnlyDataBook orderBook) throws IOException {
+        saveOrderBook(orderBook, filePath);
+    }
+
```
Comment (zhixianggg) [inline src/main/java/seedu/address/storage/JsonOrderBookStorage.java]: can it be changed to ReadOnlyDataBook `<Order>` ?

File: src/main/java/seedu/address/storage/JsonOrderBookStorage.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    /**
+     * Similar to {@link #saveOrderBook(ReadOnlyDataBook)}.
+     *
+     * @param filePath location of the data. Cannot be null.
+     */
+    public void saveOrderBook(ReadOnlyDataBook orderBook, Path filePath) throws IOException {
+        requireNonNull(orderBook);
```
Comment (zhixianggg) [inline src/main/java/seedu/address/storage/JsonOrderBookStorage.java]: same as ^

File: src/main/java/seedu/address/storage/OrderBookStorage.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    Path getOrderBookFilePath();
+
+    /**
+     * Returns OrderBook data as a {@link seedu.address.model.ReadOnlyDataBook}.
+     *   Returns {@code Optional.empty()} if storage file is not found.
+     * @throws DataConversionException if the data in storage is not in the expected format.
+     * @throws IOException if there was any problem when reading from the storage.
+     */
+    Optional<ReadOnlyDataBook> readOrderBook() throws DataConversionException, IOException;
+
```
Comment (zhixianggg) [inline src/main/java/seedu/address/storage/OrderBookStorage.java]: can it be changed to ReadOnlyDataBook `<Order>` ?

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 9]
```diff
@@ -74,4 +78,33 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) thro
...
+    // ================ OrderBook methods ==============================
+
+    @Override
+    public Path getOrderBookFilePath() {
+        return orderBookStorage.getOrderBookFilePath();
+    }
+
+    @Override
+    public Optional<ReadOnlyDataBook> readOrderBook() throws DataConversionException, IOException {
+        return readOrderBook(orderBookStorage.getOrderBookFilePath());
```
Comment (zhixianggg) [inline src/main/java/seedu/address/storage/StorageManager.java]: can it be changed to ReadOnlyDataBook `<Order>` ?

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 10]
```diff
@@ -74,4 +78,33 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) thro
...
+    }
+
+    @Override
+    public Optional<ReadOnlyDataBook> readOrderBook(Path filePath) throws DataConversionException, IOException {
+        logger.fine("Attempting to read data from file: " + filePath);
```
Comment (zhixianggg) [inline src/main/java/seedu/address/storage/StorageManager.java]: ^

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 11]
```diff
@@ -74,4 +78,33 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) thro
...
+    @Override
+    public void saveOrderBook(ReadOnlyDataBook orderBook) throws IOException {
+        saveOrderBook(orderBook, orderBookStorage.getOrderBookFilePath());
+    }
+
+    @Override
+    public void saveOrderBook(ReadOnlyDataBook orderBook, Path filePath) throws IOException {
+        logger.fine("Attempting to write to data file: " + filePath);
+        orderBookStorage.saveOrderBook(orderBook, filePath);
+    }
```
Comment (zhixianggg) [inline src/main/java/seedu/address/storage/StorageManager.java]: same as ^

**Pull request closed without merging**